### PR TITLE
Eventbrite, Pinterest Blocks: Use new shared function to test embeddable URLs

### LINKED ISSUE #14439: Eventbrite/Pinterest Blocks: Extract resolveRedirect as shared utility function
As a follow up to #14075 (see also https://github.com/Automattic/jetpack/pull/14075#discussion_r369121237), we should extract the `resolveRedirect` and `resolveUrl` functions, respectively from the Pinterest and Eventbrite blocks, and move them into a single shared utility function.

### LINKED ISSUE #14896: Eventbrite and Pinterest: Use the withNotices HOC
In https://github.com/Automattic/jetpack/pull/14884 we updated some of our blocks to use the `withNotices` HOC. We should also update the Eventbrite and Pinterest blocks to use this pattern.

### PR DESCRIPTION
**Pull request by Copons**
Fixes #14439
Fixes #14896

#### Changes proposed in this Pull Request:

* Use the `testEmbedUrl` shared function on Eventbrite and Pinterest.
* Use `withNotices` to warn when the provided URL is not embeddable.

⚠️ **NOTE**: I've commited with `--no-verify` because of a current issue with ESLint, that throws an error when parsing template literals.

#### Testing instructions:

Eventbrite: 
- Correct: https://www.eventbrite.com/e/a8c-test-event-tickets-90126734489
- Wrong: https://www.eventbrite.com/e/a8c-test-event-tickets-90126734489000

Pinterest:
- Correct: https://www.pinterest.com/anapinskywalker/
- Wrong: https://www.pinterest.com/anapinskywalker000/

* Perform the same tests with the provided URLs for both the Eventbrite and Pinterest blocks.
* Add a block and try embedding the correct URL: make sure it's embedded correctly.
* Add a block and try embedding the wrong URL: make sure it reverts to placeholder, with a notice offering to convert the URL to a link. Also make sure the conversion to link works as expected.
* Try pasting the correct URL in the editor: make sure it's converted into its block, and successfully embedded.
* Try pasting the wrong URL: make sure it's converted into its block, but to placeholder state with the error notice.
* Only for Eventbrite:
  * Add two blocks, both with the correct URL. Turn one of the blocks into a normal button (embed style "modal").
  * Save and reload.
  * Make sure the "embedding" spinner only shows up for the actual embedded version of the block, while the button/modal version appears immediately without loading.

#### Proposed changelog entry for your changes:

* N/A (mostly behind the scene changes)

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
This looks good to me. I rebased to address some unrelated issues with e2e tests that were causing Travis tests to fail.

**Comment by jeherve** [comment]
It looks like the Pinterest and EventBrite tests are still failing:
https://travis-ci.org/github/Automattic/jetpack/jobs/669750106?utm_medium=notification&utm_source=github_status

**Comment by glendaviesnz** [comment]
@Copons & @jeherve  - I fixed the e2e tests - it mostly needed a wait adding to allow the url to resolve - but also resolving the url seemed to changed the embed on the frontend for pinterest, so selector for that needed changing. 

I also modified the pinterest one to allow the pinId to be passed into the page object to match the way the newer eventbrite one was working while I was at it.

File: extensions/blocks/eventbrite/edit.js [hunk 1]
```diff
@@ -38,106 +40,78 @@ class EventbriteEdit extends Component {
...
 
-	resolveUrl = () => {
-		const { url } = this.props.attributes;
+		const newAttributes = {
+			eventId: eventIdFromUrl( url ),
+			url,
+		};
 
-		this.setState( { resolvedStatusCode: null } );
+		testEmbedUrl( newAttributes.url, this.setIsResolvingUrl )
```
Comment (glendaviesnz) [inline extensions/blocks/eventbrite/edit.js:65]: for a follow up PR at this stage, but I wonder if we convert this to a functional component if we can abstract this further with a custom hook?

Comment (Copons) [reply extensions/blocks/eventbrite/edit.js:65]: Oh, that's interesting, I've never created a custom hook, so I didn't think of doing something like that!

**Comment by Copons** [comment]
@jeherve Thanks to @glendaviesnz's change Travis is now passing! 🎉

**Review by jeherve — APPROVED** [review]

**Comment by Copons** [comment]
r205396-wpcom

**Pull request merged**